Introduce WP create button directive

furinvader opened the pull request:
Moving the create button to an appropriate directive.
This also fixes the inconsistent drop down opening behaviour.

Related to
https://community.openproject.org/work_packages/16364/activity
https://community.openproject.org/work_packages/21651/activity

furinvader committed:
Move WP create-button to a proper directive

furinvader committed:
Isolate scope of WP create button

furinvader committed:
Set disabled property of WP create button on model auth update

furinvader committed:
Add controller properties to drop down menu scope

furinvader committed:
Add stateName property to WP create button

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-refactoring-wp-create-button.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://furinvader-refactoring-wp-create-button.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Revert breakting changes

furinvader committed:
Activate WP create button if duplication is possible

If there is the case, that duplication is
possible while creation is not, this is not
a solution, although that case would not be sensible.

furinvader committed:
Attach WP create drop down menu to WP create button

Using !important is kind of a felony, but
we have to roll with it. I am sorry.

furinvader committed:
Implement isDisabled() method for consistency

furinvader commented:
I think I have to put back in some code I would have liked to remove.
Hopefully, that will make the tests pass.

oliverguenther committed:
Avoid using ProjectService when projectIdentifier is null

oliverguenther committed:
Avoid internal error when project not found
